src/navbar/navbar.tsx

```diff
@@ -179,52 +210,107 @@ const Component = <T,>(
-                        data-testid="main__brand"
-                    />
-                    {!hideNavElements && (
-                        <NavElementsContainer>
-                            <NavbarItems
-                                items={items.desktop}
-                                onItemClick={handleNavItemClick}
-                                selectedId={selectedId}
-                            />
-                            <NavbarActionButtons
-                                actionButtons={
-                                    actionButtons && actionButtons.desktop
-                                }
-                                onActionButtonClick={handleActionButtonClick}
+                <NavElementsMobileContainer>
+                    <Nav compress={compress}>
+                        <NavBrandContainer>
+                            <Brand
+                                resources={resources}
+                                compress={compress}
+                                onClick={handleBrandClick}
+                                data-testid="main__brand"
                             />
-                            <MobileMenuButton
-                                aria-label="Open nav menu"
-                                data-testid="button__mobile-menu"
-                                onClick={handleMobileMenuButtonClick}
-                                focusHighlight={false}
-                            >
-                                <MobileMenuIcon />
-                            </MobileMenuButton>
-                        </NavElementsContainer>
+                            {resourcesSecondaryBrand && (
+                                <NavLogoContainer>
+                                    <NavSeprator />
+                                    <Brand
+                                        resources={resourcesSecondaryBrand}
+                                        compress={compress}
+                                        onClick={handleSecondaryBrandClick}
+                                        data-testid="main__brand"
+                                    />
+                                </NavLogoContainer>
+                            )}
+                        </NavBrandContainer>
+
+                        {!hideNavElements && (
+                            <NavElementsLeftContainer>
+                                <NavElementsContainer>
+                                    {!toggleheaderClick && (
+                                        <NavbarItems
+                                            items={items.desktop}
+                                            onItemClick={handleNavItemClick}
+                                            selectedId={selectedId}
+                                        />
+                                    )}
+
+                                    <NavElementsRightContainer>
+                                        {children}
```

Comment:
what is this `children` intended for? I would prefer a named prop because `children` can be ambiguous